### DISCUSSION TITLE
Error when opening an MCAP file larger than 3GB

### DISCUSSION THREAD

**Discussion by hedgehog319** [foxglove/community#947] [Q&A] [answered]
Hello, there is such a problem: MCAP files up to 3 GB are freely opened, but when opening files over 3 GB I get the error **Unable to open unindexed MCAP file; unindexed files are limited to 1GB**. The files are recorded the same way so I can't figure out what the error is.

**Comment by jtbandes** [comment]
Can you please share the output of `mcap doctor` and `mcap info` run on your file? Have you tried including indexes when creating the file and see if that helps?

**Reply by hedgehog319** [reply, marked as answer]
I used `mcap doctor` on files that didn't open, interestingly they were all 3-5 GB and all were corrupted (**Failed to read next token: MCAP truncated in chunk (0x6) record content with expected length 3687693, data ended after 1668166 bytes**). Now I have made another record on **7.4 GB**, this time the file was recorded correctly and opened without errors.

It looks like some kind of strange bug, because both the corrupted and the correct files were written in a similar way:
- launch of the robot software 
- start recording the MAP file with the `ros2 bag record -s mcap -a` command

**Reply by jtbandes** [reply]
File corruption can happen when the recording process doesn't shut down cleanly. You can use `mcap recover` to create a complete file from the truncated file.
